### PR TITLE
Update Alibi-Detect wrapper and drift and outlier examples

### LINKED ISSUE #2551: new knative filtering
We're using [sourceAndType filters](https://github.com/SeldonIO/seldon-core/blob/master/components/outlier-detection/cifar10/trigger.yaml#L8) but they were [deprecated](https://knative.dev/v0.15-docs/reference/eventing/eventing-contrib-api/#eventing.knative.dev/v1alpha1.TriggerFilter) and have now been removed. Seems like the [new way is more powerful](https://github.com/knative/eventing/blob/master/docs/broker/filtering.md)

### PR DESCRIPTION
**Pull request by cliveseldon**
**What this PR does / why we need it**:

 * Update Alibi detect wrapper 
    * Ensures cloud events v1 used
    * Fixes Dockerfile USER
 * Update drift and outlier examples for knative 0.18 and istio 1.7

**Which issue(s) this PR fixes**:

Fixes #2551 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
```

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **cliveseldon**
You can assign the PR to them by writing `/assign @cliveseldon` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Wed Nov 11 15:00:48 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2626/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2626 --build=1

**Comment by seldondev** [comment]
Wed Nov 11 15:00:53 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2626/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2626 --build=2

**Comment by seldondev** [comment]
Wed Nov 11 15:03:07 UTC 2020
The logs for [lint] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2626/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2626 --build=4

**Comment by seldondev** [comment]
Wed Nov 11 15:03:56 UTC 2020
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2626/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2626 --build=3

File: components/drift-detection/cifar10/cifar10_drift.ipynb
Comment (ryandawsonuk) [inline components/drift-detection/cifar10/cifar10_drift.ipynb:39]: I notice this notebook still says "Tested on GKE with Knative 0.13 and Istio 1.3.1". We should change that to 0.18 and 1.7

**Review by ryandawsonuk — CHANGES_REQUESTED** [review]

File: components/drift-detection/cifar10/cifar10_drift.ipynb
Comment (cliveseldon) [reply components/drift-detection/cifar10/cifar10_drift.ipynb:39]: Done

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | ecb584085671c54924beac8520297ef2bcd90d01 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@cliveseldon: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | ecb584085671c54924beac8520297ef2bcd90d01 | [link]() | `/test this`
lint | ecb584085671c54924beac8520297ef2bcd90d01 | [link]() | `/test lint`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Fri Nov 13 08:33:58 UTC 2020
The logs for [pr-build] [7] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2626/7.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2626 --build=7

**Comment by seldondev** [comment]
Fri Nov 13 08:34:05 UTC 2020
The logs for [lint] [8] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2626/8.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2626 --build=8

**Review by ryandawsonuk — APPROVED** [review]
/lgtm
/approve

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[ryandawsonuk](https://github.com/SeldonIO/seldon-core/pull/2626#pullrequestreview-529915310 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [ryandawsonuk]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/2626#issuecomment-726882972 "Approved")*, *[ryandawsonuk](https://github.com/SeldonIO/seldon-core/pull/2626#pullrequestreview-529915310 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com#issuecomment-726882972/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo,ryandawsonuk]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Fri Nov 13 17:20:43 UTC 2020
The logs for [pr-build] [9] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2626/9.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2626 --build=9

**Comment by seldondev** [comment]
Fri Nov 13 17:20:57 UTC 2020
The logs for [lint] [10] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2626/10.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2626 --build=10

**Pull request merged**
